core/scripts/local/BatchGasEstimation.js

```diff
@@ -0,0 +1,208 @@
+
+  for (j = 0; j < tests.length; j++) {
+    // Binary search parameters. search starts at 64 and will either half our double if it finds
+    // it can fit that number within one block. Algorithm iterates until there is convergence
+    let requestNum = 64;
+    let converged = false;
+    let lowestFailed = 0;
+    let highestPassed = 0;
+
+    console.log("Finding maximum batch size for", tests[j], "that can be run in one transaction.");
+    while (!converged) {
+      if (lowestFailed - highestPassed == 1) {
+        console.log("*** Maximum tx per block reached for", tests[j], "as:", highestPassed, "***");
+        converged = true;
+        break;
+      }
+      console.log("testing size:", requestNum);
+      try {
+        switch (j) {
+          case 0:
+            await cycleCommit(voting, identifier, time, requestNum, registeredDerivative, voter);
+            break;
+          case 1:
+            await cycleReveal(voting, identifier, time, requestNum, registeredDerivative, voter);
+            break;
+          case 2:
+            await cycleClaim(voting, identifier, time, requestNum, registeredDerivative, voter);
+            break;
+          default:
+            break;
+        }
+        highestPassed = requestNum;
+        if (lowestFailed != 0) {
+          requestNum = Math.floor((lowestFailed + highestPassed) / 2);
+        } else {
+          requestNum = requestNum * 2;
+        }
+      } catch (error) {
+        console.log("Could not fit", requestNum, "into one transaction. Decreasing size.");
+        lowestFailed = requestNum;
+        requestNum = Math.floor((lowestFailed + highestPassed) / 2);
+      }
+      time = Math.floor(Math.random() * (await voting.getCurrentTime()));
+    }
+  }
+  console.log("Done!");
+  console.table(results);
+}
+
+const generatePriceRequests = async (requestNum, voting, identifier, time, registeredDerivative) => {
+  for (var i = 0; i < requestNum; i++) {
+    await voting.requestPrice(identifier, time + i, { from: registeredDerivative });
+  }
+};
+
+// Test how many commits can be placed within one block. Initially request n number
+// of price feeds then batch commit on them.
+const cycleCommit = async (voting, identifier, time, requestNum, registeredDerivative, voter) => {
+  // Generate prices up front for the test
+  await generatePriceRequests(requestNum, voting, identifier, time, registeredDerivative);
+
+  // Advance to commit phase
+  await moveToNextRound(voting);
+
+  const salts = {};
+  const price = getRandomSignedInt();
+  const commitments = [];
+
+  // Create the batch of commitments.
+  for (var i = 0; i < requestNum; i++) {
+    const salt = getRandomUnsignedInt();
+    const hash = web3.utils.soliditySha3(price, salt);
+    salts[i] = salt;
+    commitments[i] = { identifier: identifier, time: time + i, hash: hash, encryptedVote: [] };
+  }
+
+  // Reveal commitments.
```

Comment:
nit: comment suggests that we're revealing here, but it doesn't look like that's what's happening.